### PR TITLE
Fix lpFileName leak in InternalCreateProcess.

### LINKED ISSUE #1823: Memory leak in the CorUnix::InternalCreateProcess
This function leaks the lpFileName, it seems we will need to revert back to using an array on stack, since the function calls execve that never exits with the lpFileName as a parameter.

### PR DESCRIPTION
**Pull request by Priya91**
@janvorli 

fixes #1823

### CODE REVIEW

**Comment by janvorli** [comment]
LGTM. Btw, I have realized that the case when execve doesn't return would not even require this fix, since the execve doc says that:
`execve() does not return on success, and the text, data, bss, and stack of the calling process are overwritten by that of the program loaded.`
But it is still needed for the case when the CorUnix::InternalCreateProcess fails.

File: src/pal/src/thread/process.cpp [hunk 1]
```diff
@@ -681,13 +676,15 @@ CorUnix::InternalCreateProcess(
...
 
+    lpFileName = lpFileNamePS.OpenStringBuffer(MAX_LONGPATH-1);
     if (!getFileName(lpApplicationName, lpCommandLine, lpFileName))
     {
         ERROR("Can't find executable!\n");
         palError = ERROR_FILE_NOT_FOUND;
         goto InternalCreateProcessExit;
     }
-
+
```
Comment (adityamandaleeka) [inline src/pal/src/thread/process.cpp]: Nit: looks like this whitespace was added accidentally.

**Comment by adityamandaleeka** [comment]
LGTM modulo nit

File: src/pal/src/thread/process.cpp (hunk 1 above)
Comment (Priya91) [inline src/pal/src/thread/process.cpp]: It's a general newline after the end curly. It was not added accidentally.

Comment (adityamandaleeka) [inline src/pal/src/thread/process.cpp]: I was referring to the spaces that were added to the new line.

File: src/pal/src/thread/process.cpp [hunk 2]
```diff
@@ -681,13 +676,15 @@ CorUnix::InternalCreateProcess(
         }
     }
 
+    lpFileName = lpFileNamePS.OpenStringBuffer(MAX_LONGPATH-1);
```
Comment (sergiy-k) [inline src/pal/src/thread/process.cpp]: @Priya91, it looks like this code is missing an error check. What if OpenStringBuffer returns NULL? I think your previous PR (https://github.com/dotnet/coreclr/pull/1744) has a number of issues like this one. Could you please take a look?

Comment (joshfree) [inline src/pal/src/thread/process.cpp]: +1 here.  can you get this cleaned up in time for rc1?

Comment (Priya91) [inline src/pal/src/thread/process.cpp]: @sergiy-k Yeah sure, will take a look.

**Pull request merged**